lib/features/app/di/app_scope.dart

```diff
@@ -23,15 +25,22 @@ final class AppScope implements IAppScope {
   @override
   final ILogWriter logger;
 
+  late final IPermissionHandlerRepository _permissionHandlerRepository;
+
+  @override
+  IPermissionHandlerRepository get permissionHandlerRepository => _permissionHandlerRepository;
```

Comment:
You should not use `late final` fields. All dependencies should be injected into `AppScope`'s constructor to keep `AppScope` clean.